src/test/java/core/basesyntax/service/impl/ReportGeneratorImplTest.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ReportGeneratorImplTest {
+
+    private ReportGeneratorImpl reportGenerator;
+    private StorageServiceImpl storageService;
+
+    @BeforeEach
+    void setUp() {
+        reportGenerator = new ReportGeneratorImpl();
+        storageService = new StorageServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.clear();
+    }
+
+    @Test
+    void generateReport_validReportFormat_ok() {
+        storageService.addFruit(new Fruit("apple"), 10);
+        storageService.addFruit(new Fruit("banana"), 5);
+        String expectedReport = "fruit,quantity" + System.lineSeparator()
+                + "apple,10" + System.lineSeparator()
+                + "banana,5";
+        String actualReport = reportGenerator.generateReport();
+        Set<String> expectedLines = new HashSet<>(Arrays.asList(expectedReport
+                .split(System.lineSeparator())));
```

Comment:
Better avoid complicating your test cases. You can add expected elements to the set directly.